### ISSUE TITLE
An instance of analyzer Amazon.JSII.Analyzers.JsiiOptionalAnalyzer cannot be created

### ISSUE THREAD

**Issue by mabead** [aws/aws-cdk#5189]
When creating a dotnet cdk library project with roslyn analyzers, the analyzers fail to load.

### Reproduction Steps

1. Create a csproj file that has this content:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>library</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Amazon.CDK" Version="1.18.0" />
    <PackageReference Include="Amazon.Jsii.Analyzers" Version="*" PrivateAssets="all" />
  </ItemGroup>
  
</Project>
```
2. Compile it in Visual Studio 2019 community edition

### Error Log
Here's the output that I see in the VS Output window:

> 1>------ Build started: Project: AwsCdk.CloudWatchLogForwarder, Configuration: Debug Any CPU ------
> 1>CSC : warning CS8032: An instance of analyzer Amazon.JSII.Analyzers.JsiiOptionalAnalyzer cannot be created from C:\Users\Maxime\.nuget\packages\amazon.jsii.analyzers\0.20.8\analyzers\dotnet\cs\Amazon.JSII.Analyzers.dll : Could not load file or assembly 'System.Collections.Immutable, Version=1.2.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified..
> 1>AwsCdk.CloudWatchLogForwarder -> C:\git\AwsCdk.CloudWatchLogForwarder\src\AwsCdk.CloudWatchLogForwarder\bin\Debug\netcoreapp3.0\AwsCdk.CloudWatchLogForwarder.dll
> 1>Done building project "AwsCdk.CloudWatchLogForwarder.csproj".
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========

In other words, I can compile successfully but the Roslyn analyzers do not work.

### Environment

  - **CLI Version      : 1.18.0**
  - **Framework Version: dotnet core 3.0**
  - **OS               : windows**
  - **Language         : C#**

--- 

This is :bug: Bug Report

**Comment by RomainMuller** [comment]
Hey @mabead,

Thanks for reporting this. I think this is because of how we model the analyzer's dependencies (which I copied from a Microsoft authored blog article) might prevent the resolver to actually pull in everything that's needed when you install... And maybe this results in an analyzer that only works with very specific versions of Visual Studio.

I'm trying to confirm this hypothesis.

**Comment by RomainMuller** [comment]
I am unable to reproduce this with Visual Studio 2019 for Mac `8.3.10 (build 2)`.
It may be that Visual Studio on Windows behaves differently. I'm looking to provision a windows machine to check on there...

In the meantime, I wonder what version of `Amazon.Jsii.Analyzers` was resolved... On VS:Mac, it did not behave according to what the documentation tells me (latest known version) and instead resolved to **oldest** version... If you have gotten the same problem, maybe you want to look into forcing the version to `0.20.8` instead of using `*`. This could be related to NuGet/Home#8432.

**Comment by RomainMuller** [comment]
I've also passed #5193 which may help getting a reliably correct set-up with the `cdk init` command, given the above-mentioned hazards of using `*` version specifiers.

**Comment by RomainMuller** [comment]
Update: I got a windows machine and I can definitely reproduce with VS Community 2019. This is weirding me out a little, but at least I have something to work from now!

**Comment by mabead** [comment]
@RomainMuller Note that I have the same problem at work with Visual Studio Professional 2019 version 16.3.10.

**Comment by RomainMuller** [comment]
Yeah that appears to be a Visual Studio issue... But since other Roslyn analyzers are working, there must be a way that we can make this work... What is this way however?

Found aspnet/Mvc#5539 to look very similar to the issue at hand.

**Comment by RomainMuller** [comment]
So folks on the roslyn Gitter have told me the issue is probably that the analyzer's target is `netcoreapp3.0` whereas it'd need to be `netstandard2.0`...

**Comment by RomainMuller** [comment]
Alright the fix is at aws/jsii#1071.
